catchup/catchpointService.go

```diff
@@ -124,12 +139,14 @@ func MakeNewCatchpointCatchupService(catchpoint string, node CatchpointCatchupNo
 // Start starts the catchpoint catchup service ( continue in the process )
 func (cs *CatchpointCatchupService) Start(ctx context.Context) {
 	cs.ctx, cs.cancelCtxFunc = context.WithCancel(ctx)
+	cs.abortCtx, cs.abortCtxFunc = context.WithCancel(context.Background())
 	cs.running.Add(1)
 	go cs.run()
 }
 
 // Abort aborts the catchpoint catchup process
 func (cs *CatchpointCatchupService) Abort() {
+	cs.abortCtxFunc()
```

Comment:
please add a comment saying about cancellation func must be called before cancel cancellation func

Reply:
done.